cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -207,97 +199,116 @@ runQueryTip
+              tipTimeResult <- getProgress tipSlotNo (O.eraHistory localState) & bimap ShelleyQueryCmdPastHorizon fst & except
+
+              let tolerance = RelativeTime (secondsToNominalDiffTime 600)
+
+              return $ flip (percentage tolerance) nowSeconds tipTimeResult
+
+            mSyncProgress <- hushM syncProgressResult $ \e -> do
+              liftIO . T.hPutStrLn IO.stderr $ "Warning: Sync progress unavailable: " <> renderShelleyQueryCmdError e
+            
+            return $ Just $ O.QueryTipLocalStateOutput
+              { O.mEra = Just (O.era localState)
+              , O.mEpoch = Just epochNo
+              , O.mSyncProgress = mSyncProgress
+              }
+
+      let jsonOutput = encodePretty $ O.QueryTipOutput
+            { O.chainTip = chainTip
+            , O.mLocalState = mLocalStateOutput
+            }
 
-  let tipSlotNo = case tip of
-        ChainTipAtGenesis -> 0
-        ChainTip slotNo _ _ -> slotNo
+      case mOutFile of
+        Just (OutputFile fpath) -> liftIO $ LBS.writeFile fpath jsonOutput
+        Nothing                 -> liftIO $ LBS.putStrLn        jsonOutput
 
-  case consensusMode of
-    CardanoMode -> do
-        eResult <- liftIO $ queryEraHistoryAndSystemStart localNodeConnInfo Nothing
-        case eResult of
-          Left acqFail -> left (ShelleyQueryCmdAcquireFailure acqFail)
-          Right (eraHistory, mSystemStart) ->
-            case slotToEpoch tipSlotNo eraHistory of
-              Left e -> throwE (ShelleyQueryCmdPastHorizon e)
-              Right (epochNo, _, _) -> do
-                syncProgressResult <- runExceptT $ do
-                  systemStart <- fmap getSystemStart mSystemStart & hoistMaybe ShelleyQueryCmdSystemStartUnavailable
-                  nowSeconds <- toRelativeTime (SystemStart systemStart) <$> liftIO getCurrentTime
-                  tipTimeResult <- getProgress tipSlotNo eraHistory & bimap ShelleyQueryCmdPastHorizon fst & except
-
-                  let tolerance = RelativeTime (secondsToNominalDiffTime 600)
-                  return $ flip (percentage tolerance) nowSeconds tipTimeResult
-
-                mSyncProgress <- hushM syncProgressResult $ \e -> do
-                  liftIO . T.hPutStrLn IO.stderr $ "Warning: Sync progress unavailable: " <> renderShelleyQueryCmdError e
-
-                let output = encodePretty $ O.QueryTipOutput
-                      { O.chainTip = tip
-                      , O.era = anyEra
-                      , O.epoch = epochNo
-                      , O.syncProgress = mSyncProgress
-                      }
-
-                case mOutFile of
-                  Just (OutputFile fpath) -> liftIO $ LBS.writeFile fpath output
-                  Nothing                 -> liftIO $ LBS.putStrLn        output
     mode -> left (ShelleyQueryCmdUnsupportedMode (AnyConsensusMode mode))
 
-queryEraHistoryAndSystemStart
+queryQueryTip
```

Comment:
Let's move this to the api

Reply:
`getLocalChainTip` is still used elsewhere.

`queryQueryTip` is specific to `query tip`.  Maybe it should stay here?